extends Control

@onready var pause_mode: Node2D = %PauseMode

func _on_new_play_pressed() -> void:
	get_tree().change_scene_to_file("res://scenes/game.tscn")

func _on_quit_pressed() -> void:
	get_tree().quit()


func _on_continue_pressed() -> void:
	LoadMeneger.load_game()


func _on_statistic_pressed() -> void:
	$VBoxContainer.visible = false
	$Back.visible = true
	$Stat.visible = true


func _ready():
	# ÐŸÐ¾ÐºÐ°Ð·Ñ‹Ð²Ð°Ñ‚ÑŒ ÐºÐ½Ð¾Ð¿ÐºÑƒ Continue Ñ‚Ð¾Ð»ÑŒÐºÐ¾ ÐµÑÐ»Ð¸ Ñ„Ð°Ð¹Ð» ÑÑƒÑ‰ÐµÑÑ‚Ð²ÑƒÐµÑ‚
	$VBoxContainer/Continue.visible = LoadMeneger.load_save_file()

	var text := ""
	text += get_coins_text()
	text += "\n" + get_deaths_text()
	text += "\n" + get_times_text()

	$Stat/InfoLabel.text = text


func get_coins_text() -> String:
	var data = Db.get_all_coins_info()
	var result = "ðŸ’° ÐœÐ¾Ð½ÐµÑ‚Ñ‹:\n"
	for row in data:
		result += "- %s: %d Ð¼Ð¾Ð½ÐµÑ‚\n" % [row["username"], row["amount"]]
	return result

func get_deaths_text() -> String:
	var data = Db.get_all_death_info()
	var result = "ðŸ’€ Ð¡Ð¼ÐµÑ€Ñ‚Ð¸:\n"
	for row in data:
		result += "- %s: ÐŸÐ°Ð´ÐµÐ½Ð¸Ñ: %d, ÐœÐ¾Ð±Ñ‹: %d\n" % [row["username"], row["fall_deaths"], row["mob_deaths"]]
	return result

func get_times_text() -> String:
	var data = Db.get_all_best_times()
	var result = "â±ï¸ Ð›ÑƒÑ‡ÑˆÐµÐµ Ð²Ñ€ÐµÐ¼Ñ:\n"
	for row in data:
		result += "- %s (ÑƒÑ€Ð¾Ð²ÐµÐ½ÑŒ %d): %.2f ÑÐµÐº\n" % [row["username"], row["level_id"], row["best_time"]]
	return result


func _on_back_pressed() -> void:
	$Back.visible = false
	$VBoxContainer.visible = true
	$Stat.visible = false
